### PR TITLE
separate module from bin

### PR DESCRIPTION
**Pull request by jarofghosts**
Aaaand last PR I'll bother you with (today anyway!). This clears a path towards #7 
Interested in any thoughts about this approach!

### CODE REVIEW

**Comment by jarofghosts** [comment]
I'm actually considering refactoring to match the api suggestion in the mentioned issue.. don't know if you have a preference either way

File: readmetree.js [hunk 1]
```diff
@@ -11,78 +8,91 @@ var glob = require('glob')
...
+          callback(null, Object.keys(self.content).length)
+        }
       }
     }
   }
 }
+
+ReadmeTree.prototype.serveRequest = ReadmeTree$serveRequest
+
+function ReadmeTree$serveRequest(req, res) {
```
Comment (DTrejo) [inline readmetree.js:92]: I don't understand this separation? Why not just tack it to the prototype?

Comment (jarofghosts) [inline readmetree.js:92]: i wanted the function to be named for better stack traces, and that makes the line obscenely long

**Comment by DTrejo** [comment]
I'm cool with separating the bin; I don't know that the using the API will be all that useful? Does anyone care to make other tools for displaying mass amounts of readmes? (I suppose the answer is yes... hmm, okay sure implement that API it's whatever :)

**Comment by DTrejo** [comment]
PS Again, looks good to merge & publish

File: readmetree.js (hunk 1 above)
Comment (DTrejo) [inline readmetree.js:92]: Ah, I understand; feel free to name it "onRequest" if that helps?

D

**Comment by jarofghosts** [comment]
cool! will just rename/shorten the prototype functions and merge/publish

File: readmetree.js (hunk 1 above)
Comment (jarofghosts) [inline readmetree.js:92]: will do

**Pull request closed without merging**